File: languages/rust/experiments/java_rust/optionals.java
import java.util.Optional;

class Optionals {
    public static void main(String[] args) {
        WorkItem item1 = null;
        System.out.println("item1: " + item1);

        Optional<WorkItem> item2 = Optional.of(new WorkItem());
        System.out.println("item2: " + item2);

        Optional<WorkItem> item3 = Optional.empty();
        System.out.println("item3: " + item3);

    }
}

class WorkItem {

}

File: languages/rust/experiments/java_rust/optionals.rs
fn main() {
    let item1: Nullable<WorkItem> = Nullable::Null;
    println1("item1: {:?}", item1);

    let item2: Option<WorkItem> = Option::None;
    let item3: Option<WorkItem> = None;
    let item4: Option<WorkItem> = Some(WorkItem {});
}

#[derive(Debug)]
struct WorkItem {}

#[derive(Debug)]
enum Nullable<T> {
    Null,
    Value(T),
}
